jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/DeploymentScanner.java

```diff
@@ -275,20 +275,20 @@ public DeploymentScanner(
         this.server = Objects.requireNonNull(server);
         this.deployer = deployer == null ? server.getBean(Deployer.class) : deployer;
         installBean(deployer);
-        this.filenameFilter = Objects.requireNonNullElse(filter, new MonitoredPathFilter(monitoredDirs));
-        installBean(new DumpableCollection("monitored", monitoredDirs));
+        this.filenameFilter = Objects.requireNonNullElse(filter, new WebappsPathFilter(webappDirs));
+        installBean(new DumpableCollection("webappDirs", webappDirs));
     }
 
     /**
      * @param dir Directory to scan for deployable artifacts
      */
-    public void addMonitoredDirectory(Path dir)
+    public void addWebappsDirectory(Path dir)
```

Comment:
I'd frankly remove this and keep just the setter with a list.

Reply:
This is more in line with how we have historically done this than the set of a list (which we have never done from XML).
Allowing add also means a supplementary XML to just add another webapps directory is now possible (something that wasn't possible with older versions)